### ISSUE TITLE
ConfigMapRollout should retain the last used configmap

### ISSUE THREAD

**Issue by aylei** [pingcap/tidb-operator#603]
## Bug Report

**What version of Kubernetes are you using?**

1.12.6

**What version of TiDB Operator are you using?**

latest

**What did you do?**

Deploy a cluster with configmap rollout disabled, then enable configmap rollout.

**What did you see instead?**
```
Warning  FailedMount  27s (x10 over 4m37s)  kubelet, gke-benchmark-190621-default-pool-ca620099-brgp  MountVolume.SetUp failed for volume "startup-script" : configmaps "bench-tikv" not found
```
At first, tikv-0 is pending because the PV has not bean created, then configmap rollout enabled, but tikv-0 is still pending so it cannot be updated to the new revision. Then the PV created, tikv-0 tries to start but the configmap of old revision is not existed.

We should ensure that ConfigMapRollout retains the last used configmap.

**Comment by DanielZhangQD** [comment]
For this case, we may have to retain all versions, we cannot know exactly how many times the cm has been rolled out before PV  ready.

**Comment by weekface** [comment]
https://github.com/pingcap/tidb-operator/pull/638

**Comment by aylei** [comment]
Hopefully, this will be addressed in #986

**Comment by aylei** [comment]
closed via #1121
